File: src/clojure/Problem002.clj
(defn problem-2 [upper]
  (loop [sum 0 prev 1 n 2]
    (if (<= n upper)
      (recur (if (zero? (rem n 2)) (+ sum n) sum) 
             n
             (+ prev n))
      sum)))

(println (problem-2 4000000))

File: src/php/Problem002.php
<?php

function problem002($upper) {
	$sum = 0;
	$prev = 1;
	$n = 2;
	while ($n <= $upper) {
		if ($n % 2 == 0) {
			$sum += $n;
		}
		$tmp = $n;
		$n = $prev + $n;
		$prev = $tmp;
	}
	return $sum;
}

$result = problem002(4000000);
echo "$result\n";